### PR TITLE
Flow 2.0 design - Upload flow YAML files to DB.

### PR DESCRIPTION
**Pull request by jamiesjc**
1. Add DB operation to get latest flow version from project_flow_files table.
2. Upload flow YAML files to DB table during uploading project zip.

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java [hunk 1]
```diff
@@ -209,8 +212,13 @@ private void persistProject(final Project project, final FlowLoader loader, fina
         this.projectLoader.uploadProjectProperties(project, directoryFlowLoader.getPropsList());
 
       } else if (loader instanceof DirectoryYamlFlowLoader) {
-        // Todo jamiesjc: upload yaml file to DB as a blob
-
+        final File[] flowFiles = projectDir.listFiles(new SuffixFilter(Constants.FLOW_FILE_SUFFIX));
+        for (final File file : flowFiles) {
+          final int newFlowVersion = this.projectLoader
+              .getLatestFlowVersion(project.getId(), newProjectVersion, file.getName()) + 1;
+          this.projectLoader
```
Comment (chengren311) [inline azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java]: as discussed, maybe we should persist the name of the uploader as well.

File: azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java [hunk 2]
```diff
@@ -209,8 +212,13 @@ private void persistProject(final Project project, final FlowLoader loader, fina
         this.projectLoader.uploadProjectProperties(project, directoryFlowLoader.getPropsList());
 
       } else if (loader instanceof DirectoryYamlFlowLoader) {
-        // Todo jamiesjc: upload yaml file to DB as a blob
-
+        final File[] flowFiles = projectDir.listFiles(new SuffixFilter(Constants.FLOW_FILE_SUFFIX));
+        for (final File file : flowFiles) {
```
Comment (chengren311) [inline azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java]: one line?
for (final File file : projectDir.listFiles(new SuffixFilter(Constants.FLOW_FILE_SUFFIX)))

File: azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java (hunk 1 above)
Comment (jamiesjc) [reply azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java]: The name of uploader is already persisted in project_versions DB table. See azkaban.project.ProjectLoader#changeProjectVersion.

File: azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java (hunk 2 above)
Comment (jamiesjc) [reply azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java]: Will make the change together with next PR.

**Comment by chengren311** [comment]
Can we upload in-memory object to db instead of the raw file? Then when reading flow config from db, we don't need to do all of these processing(download to a temp file, converting the temp file to an object, then delete the temp file).

**Pull request merged**